[SIL] Enabled printing canonical module.

nate-chandler opened the pull request:
To print the module, use the new llvm flag -sil-print-guaranteed-module which parallels the existing flag -sil-view-guaranteed-cfg.  When that flag is passed, the new pass ModulePrinter is added to the diagnostic pass pipeline after mandatory diagnostics have run.  The new pass just prints the module to stdout.

nate-chandler committed:
[SIL] Enabled printing canonical module.

To print the module, use the new llvm flag -sil-print-canonical-module
which parallels the existing flag -sil-view-canonical-cfg.  When that
flag is passed, the new pass ModulePrinter is added to the diagnostic
pass pipeline after mandatory diagnostics have run.  The new pass just
prints the module to stdout.

nate-chandler commented:
@swift-ci please test

atrick reviewed (approved):
LGTM. Although you might reconsider the name. If you really want to print at the transition to canonical sil stage, then I would call it (and the view option) -sil-print-canonical-module

nate-chandler committed:
[SIL] Renamed flag for viewing canonical cfg.

Previously, the flag was named -sil-view-guaranteed-cfg.  Here, it's
renamed to -sil-view-canonical-cfg.

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@atrick Changed the wording.  So the preexisting flag is now `sil-view-canonical-cfg` and the added flag is `sil-print-canonical-module`.  Is that what you had in mind?

nate-chandler commented:
@swift-ci please test windows platform

nate-chandler commented:
@swift-ci please test windows platform

nate-chandler commented:
@swift-ci please test windows platform
